app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -77,10 +73,58 @@ public static SessionStore get() {
         GeckoSession mSession;
     }
 
+    private static class GeckoError {
+
+        public static final String ERROR_URL = "resource://android/assets/html/error/netError.xhtml";
+
+        public static final String NET_ERROR_TYPE = "about:neterror";
```

Comment:
`Server Not Found` pages are handled correctly it seems, but the error text/page is different from Firefox's:

example: http://www.asdfadsfasdfadsfadsfasdfwerqer.com:

![www asdfadsfasdfadsfadsfasdfwerqer com](https://user-images.githubusercontent.com/203725/43291667-1d7b3e9e-90e7-11e8-9fb8-8a5a3a5e0f22.png)
<img width="1530" alt="404_screen shot 2018-07-26 at 2 43 16 pm" src="https://user-images.githubusercontent.com/203725/43291674-2970a02c-90e7-11e8-9dc1-a35f26f1be1b.png">

Reply:
if I try to access `https://` for a `http://` server, it does show the expected error page:

example: https://localhost:8080 (which is a plain HTTP, non-HTTPS, server):

![image](https://user-images.githubusercontent.com/203725/43291761-7b5e305c-90e7-11e8-9fd3-41fb73823c00.png)

in Firefox desktop, this appears as a `Problem loading page` / `Secure Connection Failed` page.

![image](https://user-images.githubusercontent.com/203725/43291756-755db70e-90e7-11e8-8cc9-8847b7a3320c.png)